plasma_framework/contracts/src/exits/payment/controllers/PaymentProcessInFlightExit.sol

```diff
@@ -173,12 +173,12 @@ library PaymentProcessInFlightExit {
         private
     {
         // we flag _all_ inputs no matter it is piggybacked or not
-        // if exiting from output, all inputs are consider spent and can only exit from output furthre on.
-        // if exiting from input, for simplicity, we force all users to piggyback the input together at once
-        // instead of re-start the IFE and re-piggyback their input.
+        // if exiting from output, all inputs are consider spent and can only exit from output further on.
+        // if exiting from input, for simplicity, we force users to piggyback inputs together
+        // instead of restarting the IFE and then piggybacking their input.
         uint256 inputNumOfTheToken;
         for (uint16 i = 0; i < MAX_INPUT_NUM; i++) {
-            if (exit.inputs[i].token == token) {
+            if (exit.inputs[i].token == token && exit.inputs[i].outputId != bytes32(0)) {
```

Comment:
potentially we can have a function `isEmptyInput` in `PaymentInFlightExitModelUtils`. Meanwhile, would it be better to use amount > 0 instead? I remember we have a case that we put in the WithdrawData without `outputId` and set it in `piggyback` so with outputId or not is not so "empty". But I might be in the wrong impression that we are checking empty. 

anyway, this might only make it less clear why we are doing this so I will leave this to you.

Reply:
Good catch with `amount` instead of `outputId`. I decided not to introduce `isEmptyInput`. If you think we need it, let me know and I'll code it.